### PR TITLE
fix(workspace): preserve remote POSIX paths without macOS synthetic resolution

### LINKED ISSUE #7131: Remote terminal workspace fails on macOS when remote cwd is a Linux path (/home/<user>)
### Summary

When Hermes WebUI runs on a macOS host and is configured with a remote terminal profile (e.g., SSH backend to a Linux machine with `terminal.cwd` set to `/home/developer`), switching to the profile or resolving its workspace fails with:

```text
Error: Path does not exist: /System/Volumes/Data/home/developer
```

---

### Environment & Specific Scenario

- **Host running WebUI:** macOS (Darwin, Apple Silicon / Intel)
- **Profile Backend:** Remote SSH / Docker terminal profile (`terminal.backend: "ssh"`)
- **Remote Host:** Linux (Ubuntu/Debian/Fedora/etc.)
- **Remote Working Directory (`terminal.cwd`):** Standard Linux path (e.g., `/home/developer`)

---

### Root Cause Analysis

1. In `api/workspace.py`, `_remote_terminal_workspace_candidate()` verifies that the workspace path is under `terminal.cwd` on the remote target. However, it then calls `_resolve_path(normalized_raw)` before returning.
2. In `_clean_workspace_list()`, workspace paths are canonicalized via `p = _safe_resolve(_expanduser_path(path))` before checking profile boundaries.
3. On macOS, `/home` is managed by macOS's synthetic firmlink configuration (`/System/Volumes/Data/home`). When Python's `Path.resolve()` / `os.path.realpath()` is invoked on macOS for `/home/developer`, it expands the path to `/System/Volumes/Data/home/developer`.
4. Because `/System/Volumes/Data/home/developer` does not exist on the local Mac (where local home directories live under `/Users/`), subsequent workspace validation checks fail with `Error: Path does not exist: /System/Volumes/Data/home/developer`.
5. The existing unit test suite in `tests/test_remote_terminal_workspace.py` missed this specific edge case because `REMOTE_CWD` in the test fixture was mocked with a macOS-style path (`/Users/joeyshiue`), which avoided the macOS `/home` synthetic firmlink resolution.

---

### Proposed Fix

1. In `api/workspace.py`, return `Path(normalized_raw)` directly from `_remote_terminal_workspace_candidate()` when the path is validated as a remote target-side workspace candidate, bypassing local macOS `_resolve_path()`.
2. In `_clean_workspace_list()`, check if the path is a remote terminal workspace candidate before invoking local `_safe_resolve()`.
3. Add a unit test in `tests/test_remote_terminal_workspace.py` verifying that standard Linux `/home/<user>` paths under remote terminal backends do not expand to `/System/Volumes/Data/home/<user>` on macOS.

### PR DESCRIPTION
**Pull request by alfred-rootson**
Fixes #7131

### Problem

When Hermes WebUI runs on macOS and uses a remote terminal profile (e.g., SSH backend) pointing to a Linux machine with `terminal.cwd` set to `/home/<user>`, workspace path resolution fails with:

```text
Error: Path does not exist: /System/Volumes/Data/home/<user>
```

On macOS (Darwin), `/home` is a synthetic firmlink mapped to `/System/Volumes/Data/home`. Calling Python's `_resolve_path()` / `Path.resolve()` on a target-side Linux path causes macOS to canonicalize `/home` into `/System/Volumes/Data/home`. Subsequent local filesystem checks (`Path.exists()`) fail because that path does not exist on the local macOS host.

### Solution

1. Return `Path(normalized_raw)` directly from `_remote_terminal_workspace_candidate()` once validated, bypassing local macOS `_resolve_path()`.
2. In `_clean_workspace_list()`, check `_remote_terminal_workspace_candidate(path)` before attempting local `_safe_resolve()`.
3. Add a unit test in `tests/test_remote_terminal_workspace.py` ensuring Linux home paths like `/home/developer` do not expand to `/System/Volumes/Data/home/developer` on macOS.

### Verification

Ran `pytest tests/test_remote_terminal_workspace.py` — 16/16 passed.

### CODE REVIEW

**Comment by nesquena-hermes** [comment]
## Summary

I read the complete two-file patch at head `e2ad0df2`, the full workspace module at the PR head and on `origin/master`, and the complete remote-workspace test file in both revisions. The POSIX fix is in the correct layer: once target-side containment has been checked with `PurePosixPath`, returning the normalized target path avoids host macOS firmlink resolution. The new test does not actually reproduce that distinction on the Linux CI path, however, and the patch also changes the non-POSIX fallback without coverage.

## Code reference

The intended fix is at `api/workspace.py:178-187`, and saved-workspace cleanup now consults target-side semantics first at `api/workspace.py:247-268`:

```python
if normalized_raw is not None and normalized_cwd is not None:
    ...
    if posix_candidate == posix_base or _posix_is_within(posix_candidate, posix_base):
        return Path(normalized_raw)
...
if candidate == base or _is_within(candidate, base):
    return Path(raw)
```

The first return is exactly what the reported macOS case needs. `validate_workspace_to_add()` and `resolve_trusted_workspace()` still perform the host-local probe first at `api/workspace.py:951-978` and `:809-846`, but both correctly prefer `remote_candidate` when the local probe fails.

The regression at `tests/test_remote_terminal_workspace.py:150-159` uses the target Linux home path directly. On Linux, current master resolves that path to itself, so the assertions can pass before the patch. It does not force the pre-fix macOS transformation to the synthetic target. It also does not cover the second changed return, where relative or Windows-shaped remote paths now return `Path(raw)` instead of the previously resolved `candidate`.

## Diagnosis / recommendation

Keep the absolute-POSIX branch change. Make the regression non-vacuous by monkeypatching `_resolve_path` so the target Linux home path becomes the synthetic macOS path, while `_remote_terminal_workspace_candidate()` must still return the original normalized POSIX path. Exercise `validate_workspace_to_add`, `resolve_trusted_workspace`, and `_clean_workspace_list` under that simulated host behavior.

Unless preserving raw non-POSIX target paths is independently required, leave the fallback return as `candidate` to keep this PR scoped to the reported POSIX/macOS defect. If the fallback change is intentional, add cases for a relative remote cwd and a Windows-style remote cwd, including saved-workspace cleanup.

## Verification step

The new test should fail on `origin/master` because the simulated host resolver rewrites the target, then pass only when the POSIX branch bypasses host resolution. I did not execute PR-authored code during this read-only review.

**Comment by alfred-rootson** [comment]
Thank you for the thorough review and insightful suggestions!

I have updated the PR (commit `7b72267d`) with both recommendations:

1. **Strict Scoping:** Reverted the non-POSIX / relative fallback branch in `_remote_terminal_workspace_candidate()` back to `return candidate` so this PR remains strictly focused on the POSIX/macOS synthetic resolution defect.
2. **Robust Cross-Platform Regression Test:** Updated `test_remote_terminal_linux_home_preserves_path_without_macos_synthetic_resolution` in `tests/test_remote_terminal_workspace.py` to simulate macOS firmlink resolution via `monkeypatch.setattr(workspace, "_resolve_path", fake_resolve)`.
   - Verified that on unpatched `origin/master`, this test actively fails across all platforms (Linux/macOS/Windows) with:
     `AssertionError: assert PosixPath('/System/Volumes/Data/home/developer') == PosixPath('/home/developer')`
   - With this PR applied, all 16 tests in `test_remote_terminal_workspace.py` pass cleanly.

Ready for re-review.

File: api/workspace.py [hunk 1]
```diff
@@ -183,14 +183,14 @@ def _remote_terminal_workspace_candidate(path: str | Path) -> Path | None:
         if _is_blocked_workspace_path(Path(normalized_raw), normalized_raw) or _is_blocked_workspace_path(Path(normalized_cwd), normalized_cwd):
             return None
         if posix_candidate == posix_base or _posix_is_within(posix_candidate, posix_base):
-            return _resolve_path(normalized_raw)
+            return Path(normalized_raw)
```
[the remark this answers is not in the document]
Comment (alfred-rootson) [reply api/workspace.py:193]: Addressed in commit `6199ba3e`. Updated the docstrings for `_remote_terminal_workspace_candidate` and `_clean_workspace_list` in `api/workspace.py` to explicitly document the target-side remote path preservation semantics and why host-local resolution is bypassed for remote terminal profiles.

**Comment by nesquena-hermes** [comment]
Shipping in **exp-v0.52.249** — thanks @alfred-rootson. Verified the containment boundary holds: POSIX normalization rejects `..` traversal, NUL bytes, backslash tricks, and blocked system roots before the lexical within-`terminal.cwd` check, remote detection keys off `terminal.backend` (not the requested path, so it's not spoofable), and local workspace resolution/containment is unchanged. Nice host-agnostic fix — it also helps a Linux host with a remote profile, not just macOS.
